core/modules/path/path.module

```diff
@@ -453,10 +808,875 @@ function path_entity_update(Entity $entity) {
+    $can_redirect = TRUE;
+
+    if ($_SERVER['SCRIPT_NAME'] != $GLOBALS['base_path'] . 'index.php') {
+      // Do not redirect if the root script is not /index.php.
+      $can_redirect = FALSE;
+    }
+    elseif (!empty($_POST)) {
+      // Do not redirect if this is a post request with data.
+      $can_redirect = FALSE;
+    }
+    elseif (backdrop_is_cli()) {
+      // If this is a command line request (Drush, etc), skip processing.
+      $can_redirect = FALSE;
+    }
+    elseif ((state_get('maintenance_mode', 0) || defined('MAINTENANCE_MODE')) && !user_access('access site in maintenance mode')) {
+      // Do not redirect in offline or maintenance mode.
+      $can_redirect = FALSE;
+    }
+    elseif (!$config->get('global_admin_paths') && path_is_admin($path)) {
+      // Do not redirect on admin paths.
+      $can_redirect = FALSE;
+    }
+  }
+
+  return $can_redirect;
+}
+
+/**
+ * Compare that all values and associations in one array match another array.
+ *
+ * We cannot use array_diff_assoc() here because we need to be recursive.
+ *
+ * @param $match
+ *   The array that has the values.
+ * @param $haystack
+ *   The array that will be searched for values.
+ * @return
+ *   TRUE if all the elements of $match were found in $haystack, or FALSE
+ *   otherwise.
+ */
+function redirect_compare_array_recursive($match, $haystack) {
+  foreach ($match as $key => $value) {
+    if (!array_key_exists($key, $haystack)) {
+      return FALSE;
+    }
+    elseif (is_array($value)) {
+      if (!is_array($haystack[$key])) {
+        return FALSE;
+      }
+      elseif (!redirect_compare_array_recursive($value, $haystack[$key])) {
+        return FALSE;
+      }
+    }
+    elseif ($value != $haystack[$key]) {
+      return FALSE;
+    }
+  }
+  return TRUE;
+}
+
+/**
+ * Load a language object by its language code.
+ *
+ * @todo Remove when http://drupal.org/node/660736 is fixed in Drupal core.
+ *
+ * @param $langcode
+ *   A language code. If not provided the default language will be returned.
+ * @return
+ *   A language object.
+ */
+function redirect_language_load($langcode = LANGUAGE_NONE) {
+  $languages = &backdrop_static(__FUNCTION__);
+
+  if (!isset($languages)) {
+    $languages = language_list();
+    $languages[LANGUAGE_NONE] = NULL;
+  }
+
+  return isset($languages[$langcode]) ? $languages[$langcode] : NULL;
+}
+
+/**
+ * Build the URL of a redirect for display purposes only.
+ */
+function redirect_url($path, array $options = array(), $clean_url = NULL) {
+  if (!isset($clean_url)) {
+    $clean_url = config_get('system.core', 'clean_url');
+  }
+
+  if ($path == '') {
+    $path = '<front>';
+  }
+
+  if (!isset($options['alter']) || !empty($options['alter'])) {
+    backdrop_alter('redirect_url', $path, $options);
+  }
+
+  // The base_url might be rewritten from the language rewrite in domain mode.
+  if (!isset($options['base_url'])) {
+    if (isset($options['https']) && settings_get('https', FALSE)) {
```

Comment:
I don't think this condition is correct. The "https" setting is for sharing cookies between HTTP and HTTPS sessions.

This should probably be written as:

`if (!empty($options['https']) || config_get('system.core', 'canonical_secure'))`